app/controllers/api/vcs.php

```diff
@@ -0,0 +1,1019 @@
+                    $branchName = $parsedPayload["branch"];
+                    $repositoryId = $parsedPayload["repositoryId"];
+                    $installationId = $parsedPayload["installationId"];
+                    $pullRequestNumber = $parsedPayload["pullRequestNumber"];
+                    $SHA = $parsedPayload["SHA"];
+                    $external = $parsedPayload["external"];
+
+                    // Ignore sync for non-external. We handle it in push webhook
+                    if (!$external && $parsedPayload["action"] == "synchronize") {
+                        return $response->json($parsedPayload);
+                    }
+
+                    $github->initialiseVariables($installationId, $privateKey, $githubAppId);
+
+                    $vcsRepos = $dbForConsole->find('vcsRepos', [
+                        Query::equal('repositoryId', [$repositoryId]),
+                        Query::orderDesc('$createdAt')
+                    ]);
+
+                    $createGitDeployments($github, $installationId, $vcsRepos, $branchName, $SHA, $pullRequestNumber, $external, $dbForConsole, $getProjectDB, $request);
+                } elseif ($parsedPayload["action"] == "closed") {
+                    // Allowed external contributions cleanup
+
+                    $repositoryId = $parsedPayload["repositoryId"];
+                    $pullRequestNumber = $parsedPayload["pullRequestNumber"];
+                    $external = $parsedPayload["external"];
+
+                    if ($external) {
+                        $vcsRepos = $dbForConsole->find('vcsRepos', [
+                            Query::equal('repositoryId', [$repositoryId]),
+                            Query::orderDesc('$createdAt')
+                        ]);
+
+                        foreach ($vcsRepos as $vcsRepository) {
+                            $pullRequests = $vcsRepository->getAttribute('pullRequests', []);
+
+                            if (\in_array($pullRequestNumber, $pullRequests)) {
+                                $pullRequests = \array_diff($pullRequests, [$pullRequestNumber]);
+                                $vcsRepository = $vcsRepository->setAttribute('pullRequests', $pullRequests);
+
+                                $vcsRepository = Authorization::skip(fn () => $dbForConsole->updateDocument('vcsRepos', $vcsRepository->getId(), $vcsRepository));
+                            }
+                        }
+                    }
+                }
+            }
+
+            $response->json($parsedPayload);
+        }
+    );
+
+App::get('/v1/vcs/installations')
+    ->groups(['api', 'vcs'])
+    ->desc('List installations')
+    ->label('scope', 'public')
+    ->label('sdk.auth', [APP_AUTH_TYPE_KEY])
+    ->label('sdk.namespace', 'vcs')
+    ->label('sdk.method', 'listInstallations')
+    ->label('sdk.description', '/docs/references/vcs/list-installations.md')
```

Comment:
We miss all the `.md` files that are required when we add a new services and should be located at `docs/references`